code/game/objects/structures/dropship_equipment.dm

```diff
@@ -306,7 +306,50 @@
 		linked_console.selected_equipment = src
 		to_chat(user, span_notice("You select [src]."))
 
+//////////////////////////////////// flare launcher //////////////////////////////////////
+/obj/structure/dropship_equipment/flare_launcher
+	equip_category = DROPSHIP_WEAPON
+	name = "flare launcher system"
+	desc = "A system that deploys flares stronger than the inputted flares. Fits on the weapon attach points of dropships. You need a powerloader to lift it."
+	icon_state = "flare_system"
+	dropship_equipment_flags = IS_INTERACTABLE
+	point_cost = 150
+	var/deployment_cooldown
+	var/turf/target
+	var/stored_amount = 4
+	var/max_amount = 4
+
+/obj/structure/dropship_equipment/flare_launcher/equipment_interact(mob/user)
+	if(deployment_cooldown < world.time && stored_amount > 0) //check for cooldown and inserted flares
+		deploy_flare()
+		to_chat(user, span_notice("You deploy [src], remaining flares [stored_amount]."))
+	else
+		to_chat(user, span_warning("[src] is busy, [stored_amount] flares remaining."))
+		return //prevents spamming deployment
 
+/obj/structure/dropship_equipment/flare_launcher/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(istype(I, /obj/item/explosive/grenade/flare) && stored_amount < max_amount)
+		stored_amount++
+		to_chat(user, span_notice("You insert a flare, remaining flares [stored_amount]."))
+		qdel(I)
+		return
+
+/obj/structure/dropship_equipment/flare_launcher/update_equipment()
+	if(ship_base)
+		setDir(ship_base.dir)
+		icon_state = "flare_system_installed"
+	else
+		setDir(initial(dir))
+
+/obj/structure/dropship_equipment/flare_launcher/proc/deploy_flare()
+	playsound(loc, 'sound/weapons/guns/fire/tank_smokelauncher.ogg', 40, 1)
+	deployment_cooldown = world.time + 100
+	target = get_ranged_target_turf(src, ship_base.dir, 10)
+	var/obj/item/explosive/grenade/flare/flare_to_launch = new /obj/item/explosive/grenade/flare/strongerflare(loc)
```

Comment:
var/obj/item/explosive/grenade/flare/strongerflare/flare_to_launch = new(loc)
is a shorter and more accurate way to do this